holoviews/plotting/bokeh/element.py

```diff
@@ -1405,7 +1409,10 @@ def _get_colormapper(self, eldim, element, ranges, style, factors=None, colors=N
 
         ncolors = None if factors is None else len(factors)
         if eldim:
-            if dim_name in ranges:
+            # check if there's an actual value (not np.nan)
+            if sum((~np.isnan(v) for v in self.clim)):
```

Comment:
Should use the `isfinite` utility here and below I think, that'll handle inf and None as well.